src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java

```diff
@@ -107,4 +137,90 @@ private void validateSchemaSource(Config config) {
           .forEach(v -> v.addErrorMessage(SCHEMA_SOURCE_ERR));
     }
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  private void validateSchema(Config config) {
+    List<ConfigValue> schemaSources = config.configValues().stream()
+            .filter(v -> DatagenConnectorConfig.isExplicitlySetSchemaSource(v.name(), v.value()))
+            .collect(Collectors.toList());
+
+    Generator.Builder generatorBuilder = new Generator.Builder();
+    // don't perform this validation if multiple schema sources provided
+    if (schemaSources.size() != 1) {
+      return;
+    }
+
+    ConfigValue schemaSource = schemaSources.get(0);
+
+    switch (schemaSource.name()) {
+      case DatagenConnectorConfig.QUICKSTART_CONF:
+        if (!Quickstart.configValues.contains(((String) schemaSource.value()).toLowerCase())) {
+          schemaSource.addErrorMessage(QUICKSTART_ERR);
+          break;
+        }
+        try {
+          generator = generatorBuilder
+                  .schemaStream(getClass().getClassLoader()
+                          .getResourceAsStream(Quickstart.valueOf(((String) schemaSource.value())
+                                  .toUpperCase()).getSchemaFilename()))
+                  .build();
+        } catch (SchemaParseException | IOException e) {
+          schemaSource.addErrorMessage(QUICKSTART_SCHEMA_ERR);
+        }
```

Comment:
This logic can (and probably should) be implemented as a [ConfigDef.Validator](https://kafka.apache.org/28/javadoc/org/apache/kafka/common/config/ConfigDef.Validator.html) for the quickstart property in the config class.

Reply:
I'd prefer to keep it this way since there's no point in doing these schema validations when multiple sources are set. This way we have more control on when these validations are done. Also the schema parsed here is reused by the schemaKeyField validator. If we move this logic to three separate ConfigDef.Validator implemenations we would probably need to rewrite a lot of it for the schemaKeyField validation here. What do you think?

Reply:
There actually is still benefit to doing those validations if multiple sources are set; no reason the user should have to perform two rounds of validation to learn how to correct their connector config instead of one. It's also much cleaner, easier to read, and uses APIs already provided to us by Kafka's clients library, which has been much more thoroughly tested and vetted.

As far as reimplementing this logic for the sake of validating the schema key field--shouldn't the logic for deriving a schema from the connector config already be implemented somewhere in the code base itself? Can we extract that into a common area (possibly in the config class) and then use that logic in our validation method here? Then the check should be as simple as "use shared logic to derive schema, make sure key field is in schema". Some thought might have to be put into skipping that step when a valid schema can't be derived (because there are multiple sources specified, a schema file can't be found, an embedded schema has invalid syntax, etc.) but that shouldn't be too difficult to manage cleanly.

Reply:
> no reason the user should have to perform two rounds of validation to learn how to correct their connector config instead of one

That makes sense, I've refactored this to use ConfigDef.Validator instead.